After upgrading from 3.2.1 to 3.3.0 I get a 'Error: Cannot find module 'os''

### Bug description

I have an issue where my webpack-built cli application throws an error 
`Error: Cannot find module 'os'` after upgrading Prisma from 3.2.1 to 3.3.0.

When googling I found another thread which seems to be the same issue:
https://answers.netlify.com/t/nextjs-page-with-server-side-logic-throws-error-cannot-find-module-os/46431

When looking at the build output it looks like there is something going wierd / different with the imports.

When using Prisma **3.2.1** the output for using 'supports-color' looks like this (this is the working code):

```js
// ../../node_modules/.pnpm/supports-color@7.2.0/node_modules/supports-color/index.js
var require_supports_color2 = __commonJS2((exports2, module2) => {
  "use strict";
  var os = __webpack_require__(22037);
  var tty = __webpack_require__(76224);
  var hasFlag = require_has_flag2();
  var {env} = process;
  var forceColor;
  if (hasFlag("no-color") || hasFlag("no-colors") || hasFlag("color=false") || hasFlag("color=never")) {
    forceColor = 0;
  } else if (hasFlag("color") || hasFlag("colors") || hasFlag("color=true") || hasFlag("color=always")) {
    forceColor = 1;
  }
  if ("FORCE_COLOR" in env) {
    if (env.FORCE_COLOR === "true") {
      forceColor = 1;
    } else if (env.FORCE_COLOR === "false") {
      forceColor = 0;
    } else {
      forceColor = env.FORCE_COLOR.length === 0 ? 1 : Math.min(parseInt(env.FORCE_COLOR, 10), 3);
    }
  }
  function translateLevel(level) {
    if (level === 0) {
      return false;
    }
    return {
      level,
      hasBasic: true,
      has256: level >= 2,
      has16m: level >= 3
    };
  }
  __name(translateLevel, "translateLevel");
  function supportsColor(haveStream, streamIsTTY) {
    if (forceColor === 0) {
      return 0;
    }
    if (hasFlag("color=16m") || hasFlag("color=full") || hasFlag("color=truecolor")) {
      return 3;
    }
    if (hasFlag("color=256")) {
      return 2;
    }
    if (haveStream && !streamIsTTY && forceColor === void 0) {
      return 0;
    }
    const min2 = forceColor || 0;
    if (env.TERM === "dumb") {
      return min2;
    }
    if (process.platform === "win32") {
      const osRelease = os.release().split(".");
      if (Number(osRelease[0]) >= 10 && Number(osRelease[2]) >= 10586) {
        return Number(osRelease[2]) >= 14931 ? 3 : 2;
      }
      return 1;
    }
    if ("CI" in env) {
      if (["TRAVIS", "CIRCLECI", "APPVEYOR", "GITLAB_CI", "GITHUB_ACTIONS", "BUILDKITE"].some((sign2) => sign2 in env) || env.CI_NAME === "codeship") {
        return 1;
      }
      return min2;
    }
    if ("TEAMCITY_VERSION" in env) {
      return /^(9\.(0*[1-9]\d*)\.|\d{2,}\.)/.test(env.TEAMCITY_VERSION) ? 1 : 0;
    }
    if (env.COLORTERM === "truecolor") {
      return 3;
    }
    if ("TERM_PROGRAM" in env) {
      const version = parseInt((env.TERM_PROGRAM_VERSION || "").split(".")[0], 10);
      switch (env.TERM_PROGRAM) {
        case "iTerm.app":
          return version >= 3 ? 3 : 2;
        case "Apple_Terminal":
          return 2;
      }
    }
    if (/-256(color)?$/i.test(env.TERM)) {
      return 2;
    }
    if (/^screen|^xterm|^vt100|^vt220|^rxvt|color|ansi|cygwin|linux/i.test(env.TERM)) {
      return 1;
    }
    if ("COLORTERM" in env) {
      return 1;
    }
    return min2;
  }
  __name(supportsColor, "supportsColor");
  function getSupportLevel(stream) {
    const level = supportsColor(stream, stream && stream.isTTY);
    return translateLevel(level);
  }
  __name(getSupportLevel, "getSupportLevel");
  module2.exports = {
    supportsColor: getSupportLevel,
    stdout: translateLevel(supportsColor(true, tty.isatty(1))),
    stderr: translateLevel(supportsColor(true, tty.isatty(2)))
  };
});
```

When using Prisma **3.3.0** the output for using 'supports-color' looks like this (the __require('os') fails, so would the tty import):
```js
var require_supports_color2 = __commonJS2({
  "../../node_modules/.pnpm/supports-color@7.2.0/node_modules/supports-color/index.js"(exports2, module2) {
    "use strict";
    var os2 = __require("os");
    var tty = __require("tty");
    var hasFlag = require_has_flag2();
    var { env } = process;
    var forceColor;
    if (hasFlag("no-color") || hasFlag("no-colors") || hasFlag("color=false") || hasFlag("color=never")) {
      forceColor = 0;
    } else if (hasFlag("color") || hasFlag("colors") || hasFlag("color=true") || hasFlag("color=always")) {
      forceColor = 1;
    }
    if ("FORCE_COLOR" in env) {
      if (env.FORCE_COLOR === "true") {
        forceColor = 1;
      } else if (env.FORCE_COLOR === "false") {
        forceColor = 0;
      } else {
        forceColor = env.FORCE_COLOR.length === 0 ? 1 : Math.min(parseInt(env.FORCE_COLOR, 10), 3);
      }
    }
    function translateLevel(level) {
      if (level === 0) {
        return false;
      }
      return {
        level,
        hasBasic: true,
        has256: level >= 2,
        has16m: level >= 3
      };
    }
    __name(translateLevel, "translateLevel");
    __name2(translateLevel, "translateLevel");
    function supportsColor(haveStream, streamIsTTY) {
      if (forceColor === 0) {
        return 0;
      }
      if (hasFlag("color=16m") || hasFlag("color=full") || hasFlag("color=truecolor")) {
        return 3;
      }
      if (hasFlag("color=256")) {
        return 2;
      }
      if (haveStream && !streamIsTTY && forceColor === void 0) {
        return 0;
      }
      const min2 = forceColor || 0;
      if (env.TERM === "dumb") {
        return min2;
      }
      if (process.platform === "win32") {
        const osRelease = os2.release().split(".");
        if (Number(osRelease[0]) >= 10 && Number(osRelease[2]) >= 10586) {
          return Number(osRelease[2]) >= 14931 ? 3 : 2;
        }
        return 1;
      }
      if ("CI" in env) {
        if (["TRAVIS", "CIRCLECI", "APPVEYOR", "GITLAB_CI", "GITHUB_ACTIONS", "BUILDKITE"].some((sign2) => sign2 in env) || env.CI_NAME === "codeship") {
          return 1;
        }
        return min2;
      }
      if ("TEAMCITY_VERSION" in env) {
        return /^(9\.(0*[1-9]\d*)\.|\d{2,}\.)/.test(env.TEAMCITY_VERSION) ? 1 : 0;
      }
      if (env.COLORTERM === "truecolor") {
        return 3;
      }
      if ("TERM_PROGRAM" in env) {
        const version = parseInt((env.TERM_PROGRAM_VERSION || "").split(".")[0], 10);
        switch (env.TERM_PROGRAM) {
          case "iTerm.app":
            return version >= 3 ? 3 : 2;
          case "Apple_Terminal":
            return 2;
        }
      }
      if (/-256(color)?$/i.test(env.TERM)) {
        return 2;
      }
      if (/^screen|^xterm|^vt100|^vt220|^rxvt|color|ansi|cygwin|linux/i.test(env.TERM)) {
        return 1;
      }
      if ("COLORTERM" in env) {
        return 1;
      }
      return min2;
    }
    __name(supportsColor, "supportsColor");
    __name2(supportsColor, "supportsColor");
    function getSupportLevel(stream3) {
      const level = supportsColor(stream3, stream3 && stream3.isTTY);
      return translateLevel(level);
    }
    __name(getSupportLevel, "getSupportLevel");
    __name2(getSupportLevel, "getSupportLevel");
    module2.exports = {
      supportsColor: getSupportLevel,
      stdout: translateLevel(supportsColor(true, tty.isatty(1))),
      stderr: translateLevel(supportsColor(true, tty.isatty(2)))
    };
  }
});
```
The __require function above looks like this:
```js
var __require = /* @__PURE__ */ ((x) =>  true ? __webpack_require__(91426) : 0)(function(x) {
  if (true)
    return __webpack_require__(91426).apply(this, arguments);
  throw new Error('Dynamic require of "' + x + '" is not supported');
});
``` 
and 91426
```js
/***/ 91426:
/***/ ((module) => {

function webpackEmptyContext(req) {
	var e = new Error("Cannot find module '" + req + "'");
	e.code = 'MODULE_NOT_FOUND';
	throw e;
}
webpackEmptyContext.keys = () => ([]);
webpackEmptyContext.resolve = webpackEmptyContext;
webpackEmptyContext.id = 91426;
module.exports = webpackEmptyContext;

/***/ }),
```

### How to reproduce

The setup is pretty specfic but I hope the file examples show the reason why its failing.

### Expected behavior

Native node.js imports should just import without throwing an error.

### Prisma information

The error happends whenever you have prisma 3.3.0 inside your build.

### Environment & setup

OS: Ubuntu
NodeVersion: Tried 12 and 16.
Webpack 5

### Prisma Version

3.3.0

Have the same issue with `NextJS + Netlify` hosting. Kindly provide a solution 

`"Error: Cannot find module 'os'"`

<img width="782" alt="Screenshot 2021-11-08 at 1 21 26 AM" src="https://user-images.githubusercontent.com/7336944/140659726-71a790b6-73a1-4098-88b5-5714eac686e2.png">

--- a/helpers/compile/build.ts
+++ b/helpers/compile/build.ts
@@ -67,7 +67,7 @@ const applyCjsDefaults = (options: BuildOptions): BuildOptions => ({
   // outfile has precedence over outdir, hence these ternaries
   outdir: options.outfile ? undefined : getOutDir(options),
   // we only produce typescript types on the second run (cjs)
-  plugins: [...(options.plugins ?? []), tscPlugin, replacePlugin, onErrorPlugin],
+  plugins: [replacePlugin, ...(options.plugins ?? []), tscPlugin, onErrorPlugin],
 })
 
 // because we compile tree-shaken esm to cjs, we need to replace __require

--- a/helpers/compile/build.ts
+++ b/helpers/compile/build.ts
@@ -71,7 +71,10 @@ const applyCjsDefaults = (options: BuildOptions): BuildOptions => ({
 })
 
 // because we compile tree-shaken esm to cjs, we need to replace __require
-const replacePlugin = replaceWithPlugin([[/__require\(/g, 'require(']])
+const replacePlugin = replaceWithPlugin([
+  [/__require\(/g, 'require('], // replace the util calls
+  [/var __require.*?(?=var)/gs, ''], // remove the utility
+])
 
 /**
  * Create two deferred builds for esm and cjs. The one follows the other:

--- a/helpers/compile/build.ts
+++ b/helpers/compile/build.ts
@@ -11,6 +11,7 @@ import { tscPlugin } from './plugins/tscPlugin'
 import { onErrorPlugin } from './plugins/onErrorPlugin'
 import { fixImportsPlugin } from './plugins/fixImportsPlugin'
 import { handle } from '../blaze/handle'
+import { replaceWithPlugin } from './plugins/replaceWithPlugin'
 
 export type BuildResult = esbuild.BuildResult
 export type BuildOptions = esbuild.BuildOptions & {
@@ -66,8 +67,11 @@ const applyCjsDefaults = (options: BuildOptions): BuildOptions => ({
   // outfile has precedence over outdir, hence these ternaries
   outdir: options.outfile ? undefined : getOutDir(options),
   // we only produce typescript types on the second run (cjs)
-  plugins: [...(options.plugins ?? []), tscPlugin, onErrorPlugin],
+  plugins: [...(options.plugins ?? []), tscPlugin, replacePlugin, onErrorPlugin],
 })
+
+// because we compile tree-shaken esm to cjs, we need to replace __require
+const replacePlugin = replaceWithPlugin([[/__require\(/g, 'require(']])
 
 /**
  * Create two deferred builds for esm and cjs. The one follows the other:
@@ -133,13 +137,7 @@ async function executeEsBuild(options: BuildOptions) {
 export async function build(options: BuildOptions[]) {
   return transduce.async(
     createBuildOptions(options),
-    pipe.async(
-      computeOptions,
-      addExtensionFormat,
-      addDefaultOutDir,
-      executeEsBuild,
-      watch,
-    ),
+    pipe.async(computeOptions, addExtensionFormat, addDefaultOutDir, executeEsBuild, watch),
   )
 }
 
@@ -147,9 +145,7 @@ export async function build(options: BuildOptions[]) {
  * Executes the build and rebuilds what is necessary
  * @param builds
  */
-function watch(
-  build: esbuild.BuildResult | esbuild.BuildIncremental | undefined,
-) {
+function watch(build: esbuild.BuildResult | esbuild.BuildIncremental | undefined) {
   if (process.env.WATCH !== 'true') return build
 
   // prepare the incremental builds watcher
@@ -219,8 +215,6 @@ export function run(command: string) {
 }
 
 // gets the files to be watched from esbuild
-function getWatchedFiles(
-  build: esbuild.BuildIncremental | esbuild.BuildResult | undefined,
-) {
+function getWatchedFiles(build: esbuild.BuildIncremental | esbuild.BuildResult | undefined) {
   return Object.keys(build?.metafile?.inputs ?? {})
 }

--- a/helpers/compile/build.ts
+++ b/helpers/compile/build.ts
@@ -73,7 +73,8 @@ const applyCjsDefaults = (options: BuildOptions): BuildOptions => ({
 // because we compile tree-shaken esm to cjs, we need to replace __require
 const replacePlugin = replaceWithPlugin([
   [/__require\(/g, 'require('], // replace the util calls
-  [/var __require.*?(?=var)/gs, ''], // remove the utility
+  [/typeof __require/g, 'typeof require'], // replace the util calls
+  [/var __require =.*?(?=var)/gs, ''], // remove the utility
 ])
 
 /**

--- a/helpers/compile/build.ts
+++ b/helpers/compile/build.ts
@@ -72,9 +72,8 @@ const applyCjsDefaults = (options: BuildOptions): BuildOptions => ({
 
 // because we compile tree-shaken esm to cjs, we need to replace __require
 const replacePlugin = replaceWithPlugin([
-  [/__require\(/g, 'require('], // replace the util calls
-  [/typeof __require/g, 'typeof require'], // replace the util calls
   [/var __require =.*?(?=var)/gs, ''], // remove the utility
+  [/__require(?!\w)/gs, 'require'], // replace calls to util
 ])
 
 /**

--- a/helpers/compile/plugins/externalPlugin.ts
+++ b/helpers/compile/plugins/externalPlugin.ts
@@ -1,7 +1,7 @@
 import type { Plugin } from 'esbuild'
 
 /**
- * esbuild plugin that marks all packages as external
+ * Marks all packages as external.
  */
 const externalPlugin: Plugin = {
   name: 'externalPlugin',

--- a/helpers/compile/plugins/fill-plugin/fillPlugin.ts
+++ b/helpers/compile/plugins/fill-plugin/fillPlugin.ts
@@ -128,8 +128,8 @@ function onLoad(fillers: Fillers, args: esbuild.OnLoadArgs): esbuild.OnLoadResul
 const load = loader({})
 
 /**
- * esbuild plugin that provides a simple way to use esbuild's injection
- * capabilities while providing sensible defaults for node polyfills.
+ * Provides a simple way to use esbuild's injection capabilities while providing
+ * sensible defaults for node polyfills.
  * @see https://v2.parceljs.org/features/node-emulation/
  * @see https://github.com/Richienb/node-polyfill-webpack-plugin/blob/master/index.js
  * @param fillerOverrides override default fillers

--- a/helpers/compile/plugins/fixImportsPlugin.ts
+++ b/helpers/compile/plugins/fixImportsPlugin.ts
@@ -1,5 +1,8 @@
 import type * as esbuild from 'esbuild'
 
+/**
+ * For dependencies that forgot to add them into their package.json.
+ */
 export const fixImportsPlugin: esbuild.Plugin = {
   name: 'fixImportsPlugin',
   setup(build) {

--- a/helpers/compile/plugins/onErrorPlugin.ts
+++ b/helpers/compile/plugins/onErrorPlugin.ts
@@ -1,5 +1,8 @@
 import type * as esbuild from 'esbuild'
 
+/**
+ * Causes esbuild to exit immediately with an error code.
+ */
 export const onErrorPlugin: esbuild.Plugin = {
   name: 'onErrorPlugin',
   setup(build) {

--- a/helpers/compile/plugins/replaceWithPlugin.ts
+++ b/helpers/compile/plugins/replaceWithPlugin.ts
@@ -9,6 +9,11 @@ function applyReplacements(contents: string, replacements: [RegExp, string][]) {
   return contents
 }
 
+/**
+ * Replace the contents of a file with the given replacements.
+ * @param replacements
+ * @returns
+ */
 export const replaceWithPlugin = (replacements: [RegExp, string][]): esbuild.Plugin => {
   return {
     name: 'replaceWithPlugin',

--- a/helpers/compile/plugins/replaceWithPlugin.ts
+++ b/helpers/compile/plugins/replaceWithPlugin.ts
@@ -1,0 +1,23 @@
+import type * as esbuild from 'esbuild'
+import fs from 'fs'
+
+function applyReplacements(contents: string, replacements: [RegExp, string][]) {
+  for (const [regex, replacement] of replacements) {
+    contents = contents.replace(regex, replacement)
+  }
+
+  return contents
+}
+
+export const replaceWithPlugin = (replacements: [RegExp, string][]): esbuild.Plugin => {
+  return {
+    name: 'replaceWithPlugin',
+    setup(build) {
+      build.onLoad({ filter: /.*/ }, async (args) => {
+        const contents = await fs.promises.readFile(args.path, 'utf8')
+
+        return { contents: applyReplacements(contents, replacements) }
+      })
+    },
+  }
+}

--- a/helpers/compile/plugins/tscPlugin.ts
+++ b/helpers/compile/plugins/tscPlugin.ts
@@ -1,6 +1,9 @@
 import type * as esbuild from 'esbuild'
 import { run } from '../build'
 
+/**
+ * Triggers the TypeScript compiler.
+ */
 export const tscPlugin: esbuild.Plugin = {
   name: 'tscPlugin',
   setup(build) {

--- a/packages/client/helpers/build.ts
+++ b/packages/client/helpers/build.ts
@@ -20,7 +20,11 @@ const runtimeBuildConfig: BuildOptions = {
   outfile: 'runtime/index',
   bundle: true,
   external: external,
-  define: { 'globalThis.NOT_PRISMA_DATA_PROXY': 'true' },
+  define: {
+    'globalThis.NOT_PRISMA_DATA_PROXY': 'true',
+    // that fixes an issue with lz-string umd builds
+    'define.amd': 'false',
+  },
 }
 
 // we define the config for browser

